src/k8s/kubernetes.py

```
@@ -624,6 +644,18 @@ def _create_analysis_network_policy(analysis_name: str, nginx_name: str, namespa
     network_client.create_namespaced_network_policy(namespace=namespace, body=network_body)
 
 
+def _delete_k8s_deployment(app_client: client.AppsV1Api, name: str, namespace: str) -> None:
+    try:
+        app_client.delete_namespaced_deployment(async_req=False, name=name, namespace=namespace)
+    except client.exceptions.ApiException as e:
+        if e.reason == 'Not Found':
+            logger.warning(f"Could not find deployment {name} for deletion")
+        else:
+            logger.error(f"Unknown error when attempting to delete deployment {name} (reason={e.reason})")
+    except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)